### PR TITLE
Make animation blit cache robust against 3d viewpoint changes.

### LINKED ISSUE #16069: matplotlib glitch when rotating interactively a 3d animation
### Bug report

**matplotlib glitch when rotating interactively a 3d animation**
Plotting a 3d animation, matplotlib glitches when rotating interactively, despite doing it well when zooming. Setting a callback to stop animation during rotation doesn't seem to help, since (see code and attached video) the little red ball, which should be following the blue line, goes "off-track"
Using matplotlib 3.1.2

**without callback**
no_callback

**with callback**
callback

[Videos.zip](https://github.com/matplotlib/matplotlib/files/4015531/Videos.zip)

**Code for reproduction**

```python
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from matplotlib.animation import FuncAnimation

t = np.linspace(0, 20*np.pi, 1000)
x = t
y = np.cos(t)
z = np.sin(t)

def animate(i, *fargs):
    x, y, z = fargs[0:3]
    ln = fargs[3]
    # Plot small ball at time t[i]
    ln.set_data_3d([x[i]], [y[i]], [z[i]])
    ln.set_color('r')
    ln.set_marker('.')
    ln.set_markersize(10)
    return ln,

fig = plt.figure(figsize=(9,7))
ax = plt.axes(projection='3d')
fix_ln = ax.plot(x, y, z, 'b-')[0]
ln = ax.plot([], [], [])[0]

ani = FuncAnimation(fig, animate, frames=np.arange(t.size), repeat=False, fargs=(x, y, z, ln), interval=0, blit=True)
# Comment the two following lines to play animation without callbacks:
plt.connect('button_press_event', lambda event: ani.event_source.stop())
plt.connect('button_release_event', lambda event: ani.event_source.start())
plt.show()
```

**Matplotlib version**

  * Operating system: Ubuntu 19.10 eoan, Linux 5.3.0-24-generic <span>#</span>26-Ubuntu SMP Thu Nov 14 01:33:18 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
  * Matplotlib version:  3.1.2 (installed with pip)
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.7.3
  * IPython version: 7.8.0
  * Other libraries: numpy 1.17.2

### PR DESCRIPTION
**Pull request by anntzer**
Instead of connecting to the xlim_changed/ylim_changed events to clear
the blit cache, use Axes._get_view(), which also works for 3d axes.

Closes #16069.  No test because good luck with that, but one can repro with the example at that issue.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
_get_view returns the axes limits (plus the view angles in 3D), which is something that needs to be accessed anyways for drawing the artist.  So yes it's a couple more attribute accesses but I doubt it matters (but I have not checked this).

In theory custom projections need to implement _get_view anyways to interact properly with the home/previous/next buttons, so it seems annoying to request that they additionally need to implement view_changed (_get_view and view_changed going out of sync would be yet another possible source of bugs, whereas if there's only one source of truth, it can't go out of sync with itself).

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -1186,53 +1186,52 @@ def _post_draw(self, framedata, blit):
...
-            if a in bg_cache:
-                a.figure.canvas.restore_region(bg_cache[a])
+        for ax in axes:
+            try:
+                view, bg = self._blit_cache[ax]
+            except KeyError:
+                pass
+            else:
+                if ax._get_view() == view:
+                    ax.figure.canvas.restore_region(bg)
```
Comment (tacaswell) [inline lib/matplotlib/animation.py]: Might as well evict invalid backgrounds here as well?

**Comment by tacaswell** [comment]
This seems to be simpler over all.  While this is in the hot-loop of the animation loop so we should be concerned, but on my few year laptop, `ax._get_view()` takes ~9us.  My (un-benchmarked) hunch as that we are getting back more than that with the other cleanups in this PR.

File: lib/matplotlib/animation.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/animation.py]: done, good point.

**Pull request merged**